### PR TITLE
Andrew/mondaynight backend

### PR DESCRIPTION
**Pull request by AndrewReardigan**

### CODE REVIEW

File: server/utils/apiFeatures.js [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+      const queryObj = { ...this.queryString };
+      const excludedFields = ['page', 'sort', 'limit', 'fields'];
+      excludedFields.forEach(el => delete queryObj[el]);
+  
+      // 1B) Advanced filtering
+      let queryStr = JSON.stringify(queryObj);
+      queryStr = queryStr.replace(/\b(gte|gt|lte|lt)\b/g, match => `$${match}`);
+  
+      this.query = this.query.find(JSON.parse(queryStr));
+  4
```
Comment (kcbreuil) [inline server/utils/apiFeatures.js:18]: @AndrewReardigan what is this 4?

File: package.json [hunk 2]
```diff
@@ -7,24 +7,41 @@
     "@material-ui/core": "^4.11.0",
     "axios": "^0.20.0",
     "bcrypt": "^5.0.0",
+    "bcryptjs": "^2.4.3",
```
Comment (ErnieAtLYD) [inline package.json:10]: Why bcryptjs versus bcrypt? (Not leading you one way or another, legit curious)

File: server/utils/apiFeatures.js (hunk 1 above)
Comment (ErnieAtLYD) [reply server/utils/apiFeatures.js:18]: @AndrewReardigan that MUST be a typo, right?

File: server/db/templates/emailTemplate.pug [hunk 3]
```diff
@@ -0,0 +1,313 @@
+//- Email template adapted from https://github.com/leemunroe/responsive-html-email-template
```
Comment (ErnieAtLYD) [inline server/db/templates/emailTemplate.pug:1]: Annnnnd now I understand why you asked me about pug lol

File: server/routes/parkRoutes.js [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+const router = express.Router();
+
+// router.param('id', parkController.checkID);
+
+// POST /park/234fad4/reviews
+// GET /park/234fad4/reviews
+
+router.use('/:parkId/reviews', reviewRouter);
+
+router
```
Comment (ErnieAtLYD) [inline server/routes/parkRoutes.js:15]: These are still endpoints for `/parks/`, right? So like

`/parks/:parkId/reviews`
`/parks/park-stats`
`/parks/:year`

etc?

File: package.json (hunk 2 above)
Comment (AndrewReardigan) [reply package.json:10]: Was literally me playing with them. I am going to remove what I need today out of package.json.

Comment (ErnieAtLYD) [reply package.json:10]: OK so @AndrewReardigan - I found this:
https://github.com/kelektiv/node.bcrypt.js/issues/705#:~:text=vs%2Dbrypt.js-,Short%20summary%3A%20bcrypt%20is%20a%20native%20(C%2B%2B)%20module%2C,dependencies%20to%20compile%20from%20source.

So I can be convinced that `bcryptjs` is the better way to go. My only ask is that you use `bcryptjs` _instead_ of `bcrypt`, and remove any references to `bcrypt` itself if it's being used elsewhere within the code.

File: server/utils/apiFeatures.js (hunk 1 above)
Comment (AndrewReardigan) [reply server/utils/apiFeatures.js:18]: Oh wow. It is safe to assume that 4 is not suppose to be there. Will be removed.

File: package.json (hunk 2 above)
Comment (AndrewReardigan) [reply package.json:10]: Yes Sir. Not needed packages will be removed.

File: server/routes/parkRoutes.js (hunk 4 above)
Comment (AndrewReardigan) [reply server/routes/parkRoutes.js:15]: Yes Sir. 
There are some things out of wack right now. Typos that I am discovering as well, parkID vs parkId. Anything not needed is coming out. Anything that needs to be altered will be. 

No excuses, but this is the result of a PR at 2:30am after a couple days. I will get better and be more on point. I am not moving forward with ANYTHING until I iron out all details and try my best to properly //comment them.

**Pull request merged**